pkg/prometheus/resource_selector.go

```diff
@@ -677,6 +677,24 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 			}
 		}
 
+		var scrapeInterval, scrapeTimeout monitoringv1.Duration
+
+		if sc.Spec.ScrapeInterval == nil {
+			scrapeInterval = ""
+		} else {
+			scrapeInterval = *sc.Spec.ScrapeInterval
+		}
+
+		if sc.Spec.ScrapeTimeout == nil {
+			scrapeTimeout = ""
+		} else {
+			scrapeTimeout = *sc.Spec.ScrapeTimeout
+		}
+
+		if err = validateScrapeIntervalAndTimeout(rs.p, scrapeInterval, scrapeTimeout); err != nil {
```

Comment:
(nit) I wonder if we should change the signature of that function to accept pointers instead, it would simplify the logic above. We could also push that logic in the function and have it return both values or nil.

Reply:
ya so I had initially put that change but since that updated other code blocks I thought I should do it in another change. But I just updated it again if its okay to go in this change itself :)

Reply:
It sounds good to me, thanks!